### PR TITLE
[UI] #16482 - Select All link for table-specific privileges

### LINKED ISSUE #16482: Selector for table-specific privileges should allow "select all"
When editing a user and picking table-specific privileges, each column is shown in a selector for "Select," "Insert," "Update," and "Reference" privileges.

If I click on one of these, then use my keyboard shortcut for my operating system to select all (command-a on Mac or ctrl-a on Windows) I would like all columns to be selected.

Instead, all of the text on the page is selected.

Perhaps it would be good to have a "select all" clickable link, too?

image

### PR DESCRIPTION
**Pull request by shucon**
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

### Description

Please describe your pull request.

Fixes #16482 

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

File: js/src/server/privileges.js [hunk 1]
```diff
@@ -461,6 +461,22 @@ AJAX.registerOnload('server/privileges.js', function () {
...
 
+    var tableSelectAll = function (e) {
+        e.preventDefault();
+        var id = $(this)[0].id;
+        var method = id.split('_')[0];
+        var options = $('#select_' + method + '_priv')[0].children;
+        for (var i = 0; i < options.length; i++) {
+            options[i].selected = true;
+        }
+        e.stopPropagation();
```
Comment (williamdes) [inline js/src/server/privileges.js:472]: Stop would make more sense above, right?

File: js/src/server/privileges.js [hunk 2]
```diff
@@ -461,6 +461,22 @@ AJAX.registerOnload('server/privileges.js', function () {
         addOrUpdateSubmenu();
     }
 
+    var tableSelectAll = function (e) {
+        e.preventDefault();
+        var id = $(this)[0].id;
+        var method = id.split('_')[0];
```
Comment (williamdes) [inline js/src/server/privileges.js:467]: This is terrible and will break, can use use a `data-` attribute instead?

**Review by williamdes — CHANGES_REQUESTED** [review]

File: templates/server/privileges/privileges_table.twig [hunk 3]
```diff
@@ -23,6 +23,8 @@
         {% endfor %}
       </select>
 
+      <button class="btn btn-secondary" id="select_priv_all" data-method="select" >{% trans 'Select all' %}</button><br/>
```
Comment (williamdes) [inline templates/server/privileges/privileges_table.twig:26]: Not sure about the `<br>`, @mauriciofauth do you think this is okay ?

File: js/src/server/privileges.js [hunk 4]
```diff
@@ -461,6 +461,20 @@ AJAX.registerOnload('server/privileges.js', function () {
         addOrUpdateSubmenu();
     }
 
+    var tableSelectAll = function (e) {
+        e.preventDefault();
+        var method = $(this)[0].getAttribute('data-method');
```
Comment (williamdes) [inline js/src/server/privileges.js:466]:
```suggestion
        var method = $(this).first().data('method');
```
That should work too
And avoid index errors

Comment (mauriciofauth) [inline js/src/server/privileges.js:466]:
```suggestion
        const method = e.target.getAttribute('data-method');
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

File: templates/server/privileges/privileges_table.twig [hunk 5]
```diff
@@ -44,6 +50,12 @@
         {% endfor %}
       </select>
 
+      <div>
+        <button class="btn btn-link p-0" id="insert_priv_all" type="button"data-select-target="#select_insert_priv">
```
Comment (williamdes) [inline templates/server/privileges/privileges_table.twig:54]:
```suggestion
        <button class="btn btn-link p-0" id="insert_priv_all" type="button" data-select-target="#select_insert_priv">
```

File: yarn.lock [hunk 6]
```diff
@@ -4574,10 +4574,10 @@ node-modules-regexp@^1.0.0:
...
   integrity sha1-jZ2+KJZKSsVxLpExZCEHxx6Q7EA=
 
-node-notifier@^8.0.1:
-  version "8.0.1"
-  resolved "https://registry.yarnpkg.com/node-notifier/-/node-notifier-8.0.1.tgz#f86e89bbc925f2b068784b31f382afdc6ca56be1"
-  integrity sha512-BvEXF+UmsnAfYfoapKM9nGxnP+Wn7P91YfXmrKnfcYCx6VBeoN5Ez5Ogck6I8Bi5k4RlpqRYaw75pAwzX9OphA==
+node-notifier@^8.0.0:
+  version "8.0.2"
+  resolved "https://registry.yarnpkg.com/node-notifier/-/node-notifier-8.0.2.tgz#f3167a38ef0d2c8a866a83e318c1ba0efeb702c5"
+  integrity sha512-oJP/9NAdd9+x2Q+rfphB2RJCHjod70RcRLjosiPMMu5gjIfwVnOUGq2nbTjTUbmy0DJ/tFIVT30+Qe3nzl4TJg==
```
Comment (williamdes) [inline yarn.lock:4580]: could you exclude this change ?

Comment (shucon) [reply yarn.lock:4580]: Sorry, committed that by mistake.

File: js/src/server/privileges.js [hunk 7]
```diff
@@ -461,6 +461,19 @@ AJAX.registerOnload('server/privileges.js', function () {
         addOrUpdateSubmenu();
     }
 
+    var tableSelectAll = function (e) {
+        const method = e.target.getAttribute('data-select-target');
+        var options = $(method).first().children();
```
Comment (williamdes) [inline js/src/server/privileges.js:466]: Jquery has no `.selected(true);` or `.attr('selected', true);` to avoid the loop ?

**Review by williamdes — APPROVED** [review]

File: js/src/server/privileges.js (hunk 7 above)
Comment (shucon) [reply js/src/server/privileges.js:466]: `.each()` is the closest I could find

**Review by williamdes — APPROVED** [review]
Works fine as expected

**Pull request merged**